cloudbridge/cloud/providers/gce/provider.py

```diff
@@ -157,6 +160,38 @@ def parse_url(self, url):
                 out.parameters[parameter] = m.group(index + 1)
             return out
 
+    def get_resource_url_with_default(self, resource, identifier):
+        """
+        Build a GCPResourceUrl from a resource's name and its identifier.
+        If the identifier is a valid GCP resource URL, then we build the
+        GCPResourceUrl object by parsing this URL. If the identifier is its
+        short name, then we build the GCPResourceUrl object by constructing
+        the resource URL with default project, region, zone values.
+        """
+        # If identifier is a valid GCP resource URL, then parse it.
+        if identifier.startswith(self._root_url):
+            return self.parse_url(identifier)
+        # Otherwise, construct resource URL with default values.
+        if resource not in self._resources:
+            return None
+        resource_desc = self._connection._resourceDesc['resources'][resource]
```

Comment:
For the rest of this method I prefer the following slightly simpler and more direct implementation. What do you think?

```
parsed_url = GCPResourceUrl(resource, self._connection)
for _, parameter in self._resources[resource]['parameters']:
  if parameter in DEFAULT_VALUES:
    parsed_url.parameters[parameter] = self._parameter_defaults[parameter]
  else:
    parsed_url.parameters[parameter] = name
return parsed_url
```

Define `self._parameter_defaults` in `__init__` since they are not supposed to be changed.

Reply:
Done. Thank you very much for the suggestion.